doc/neps/ufunc-overrides.rst

```diff
@@ -575,14 +578,41 @@ For option (2), the relevant example is the fourth, with ``arr *=
 mine``: if we had let the :obj:`NotImplemented` pass, python would have
 replaced this with ``arr = mine.__rmul__(arr)``, which is not wanted.
 
+Because the semantics of Ufunc overrides and Python's binary operations
+are nearly identical, in most cases options (1) and (2) will
+yield the same result with the same implementation of ``__array_ufunc__``.
+One exception is the order in which implementations are tried when the
+second argument is a subclass of the first argument, due to a Python
+bug [9]_ expected to be fixed in Python 3.7.
+
+In general, we recommend adopting option (1), which is the option most
+similar to that used by :class:`ndarray` itself. Note that option (1)
+is viral, in the sense that any other class that wishes to support binary
+operations with your class now must also follow these rules for supporting
+binary arithmetic with :class:`ndarray` (i.e., they must either implement
+``__array_ufunc__`` or set it to :obj:`None`). We believe this is a good
+thing, because it ensures the consistency of ufuncs and arithmetic on all
+objects that support them.
+
+To make implementing such array-like classes easier, the mixin class
+:class:`~numpy.lib.mixins.NDArrayOperatorsMixin` provides option (1) style
+overrides for all binary operators with corresponding ufuncs. For classes
+that wish to implement ``__array_ufunc__`` for compatible versions
+of NumPy but that also need to support binary arithmetic with NumPy arrays
+on older versions, we recommend using a backwards compatible variant of
+:class:`~numpy.lib.mixins.NDArrayOperatorsMixin` that we intend to provide
+in a separate repository.
```

Comment:
It's a good question how necessary this really is. But we already have a test suite for NDArrayOperatorsMixin, so I think it would be pretty easy to write this backport.